sceptre/cli.py

```diff
@@ -574,6 +576,170 @@ def get_stack_policy(ctx, environment, stack):
     write(response.get('StackPolicyBody', {}))
 
 
+@cli.group(name="init")
+def init():
+    """
+    Commands for initialising Sceptre projects.
+    """
+    pass
+
+
+@init.command("env")
+@click.argument('env')
+@catch_exceptions
+@click.pass_context
+def init_environment(ctx, env):
+    """
+    Initialises an environment in a project.
+
+    Creates ENVIRONMENT folder in the project.
+    """
+    cwd = ctx.obj["sceptre_dir"]
+    for item in os.listdir(cwd):
+        # If already a config folder create a sub environment
+        if os.path.isdir(item) and item == "config":
+            config_dir = os.path.join(os.getcwd(), "config")
+            _create_new_environment(config_dir, env)
+
+
+@init.command("project")
+@catch_exceptions
+@click.argument('project_name')
+@click.pass_context
+def init_project(ctx, project_name):
+    """
+    Initialises a new project.
+
+    Creates PROJECT_NAME project folder.
+    """
+    cwd = os.getcwd()
+    sceptre_folders = {"config", "templates"}
+    project_folder = os.path.join(cwd, project_name)
+    try:
+        os.mkdir(project_folder)
+    except OSError as e:
+        # Check if environment folder already exists
+        if e.errno == errno.EEXIST:
+            raise ProjectExistsError(
+                'Folder \"{0}\" already exists.'.format(project_name)
+            )
+        else:
+            raise
+
+    for folder in sceptre_folders:
+        folder_path = os.path.join(project_folder, folder)
+        os.makedirs(folder_path)
+
+    defaults = {
+        "project_code": project_name,
+        "region": os.environ.get("AWS_DEFAULT_REGION", "")
+    }
+
+    config_path = os.path.join(cwd, project_name, "config")
+    _create_config_file(config_path, config_path, defaults)
+
+
+def _create_new_environment(config_dir, new_path):
+    """
+    Creates the subfolder for the environment specified by `path` starting
+    from the `config_dir`. Even if folder path already exists, ask the user if
+    they want to initialise `config.yaml`.
+
+    :param config_dir: The directory path to the top-level config folder.
+    :type config_dir: str
+    :param path: The directory path to the environment folder.
+    :type path: str
+    """
+    # Create full path to environment
+    folder_path = os.path.join(config_dir, new_path)
+    environment_exists = False
+
+    # Make folders for the anvironment
+    try:
+        os.makedirs(folder_path)
+    except OSError as e:
+        # Check if environment folder already exists
+        if e.errno == errno.EEXIST:
+            environment_exists = True
+        else:
+            raise
+
+    init_config_msg = 'Do you want initialise config.yaml?'
+
+    if environment_exists:
+        init_config_msg = 'Environment path exists. ' + init_config_msg
+
```

Comment:
Does it make sense to just put this init_config_msg assignment inside the try/if statement instead?

Reply:
Ah right! instead of using the `environment_exists` variable?

Reply:
Yep!